File: Todos/C/BEE 3301.c
/*
  autor: Vinist021;
  data: 15/02/2025;
  nome: Sobrinho do Meio;
*/

#include <stdio.h>

int maior(int a, int b, int c)
{
	int mai = a;
	if(mai < b)
		 mai = b;
	if(mai < c)
		mai = c;
		
	return mai;
}

int menor(int a, int b, int c)
{
	int men = a;
	if(men > b)
		 men = b;
	if(men > c)
		men = c;
		
	return men;
}

int main() {
	
	int H, Z, L;
	
	scanf("%d %d %d", &H, &Z, &L);
	
	int maiorNum = maior(H, Z, L);
	int menorNum = menor(H, Z, L);
	
	if(H != maiorNum && H != menorNum)
		printf("huguinho\n");
	else if(Z != maiorNum && Z != menorNum)
		printf("zezinho\n");
	else
		printf("luisinho\n");
	
	return 0;
}

File: Todos/Python/BEE 3301.py
#   autor: Vinist021;
#   data: 15/02/2025;
#   nome: Sobrinho do Meio;

H, Z, L = map(int, input().split())
l = [H, Z, L]
l.sort()

if l[1] == H:
    print('huguinho')
elif l[1] == Z:
    print('zezinho')
else:
    print('luisinho')